app/assets/javascripts/file_manager.js

```diff
@@ -0,0 +1,197 @@
+                a.style.display = 'none';
+                document.body.appendChild(a);
+                a.click();
+                document.body.removeChild(a);
+                URL.revokeObjectURL(url);
+                console.log(`Downloaded: ${data.filename}`);
+                resolve();
+            },
+            error: function (xhr, status, error) {
+                console.error(`Failed to download ${path}: ${error}`);
+                reject(error);
+            }
+        });
+    })
+}
+
+function uploadFile(file, path, name) {
+    let formData;
+    let pathSegments = path.split('/');
+    pathSegments.pop();
+    let modifiedPath = pathSegments.join('/');
+    formData = new FormData();
+    formData.append('file', file);
+    formData.append('name', name);
+    return new Promise(function(resolve, reject) {
+        $.ajax({
+            url: modifiedPath,
+            type: "POST",
+            data: formData,
+            contentType: false,
+            processData: false,
+            success: function () {
+                console.log("Uploaded file successfully");
+                resolve();
+            },
+            error: function (xhr, status, error) {
+                console.error(`Failed to upload file: ${error}`);
+                reject(error);
+            }
+        });
+    });
+}
+
+function uploadAllFiles(path) {
+    let inputElement = document.getElementById('fileInput');
+    let files = inputElement.files;
+    const uploadPromises = [];
+    for (let i = 0; i < files.length; i++) {
+        let file = files[i];
+        uploadPromises.push(uploadFile(file, path, ""));
+    }
+    if (files.length > 0) {
+        Promise.all(uploadPromises)
+        .then(() => {
+            alert("All files uploaded successfully.");
+            location.reload();
+        })
+        .catch((error) => {
+            alert("Some files failed to upload successfully. Ensure that you are not in the root directory, the file is smaller than 1 GB, and does not already exist.");
+            location.reload();
+        });
+    } else {
+        console.log('No files selected.');
+    }
+}
+
+function getSelectedItems() {
+    let selectedItems = $("input[type='checkbox'][class='check']:checked");
+    let paths = [];
+    selectedItems.each(function(index, element) {
+        paths.push(jQuery(element).prop('value'));
+    });
+    return paths
+}
+
+function createFolder(path) {
+    let name = prompt("Name of folder: ");
+    if (name !== "" && name !== null) {
+        uploadFile("", path, name)
+            .then(() => {
+                alert("Folder created successfully.");
+                location.reload();
+            })
+            .catch((error) => {
+                alert("Failed to create folder. Check that you are not in the root directory and that the tile/folder does not already exist.");
+            })
+        ;
+    }
+}
+
+function handleDownloadClick() {
```

Comment:
Nit: for `handleDownloadClick` and `handleDeleteSelected` maybe display an alert if there are no selected items? Or even better, disable the buttons

Reply:
Addressed, buttons are disabled if nothing is selected.